Source/Diagnostics/ReducedDiags/ParticleMeanEnergy.cpp

```diff
@@ -0,0 +1,155 @@
+#include "ParticleMeanEnergy.H"
+#include "WarpX.H"
+#include "WarpXConst.H"
+#include "AMReX_REAL.H"
+#include <iostream>
+#include <cmath>
+
+/// constructor
+ParticleMeanEnergy::ParticleMeanEnergy (
+std::string rd_name, std::ofstream & ofs )
+: ReducedDiags{rd_name}
+{
+    /// get WarpX class object
+    auto & warpx = WarpX::GetInstance();
+
+    /// get MultiParticleContainer class object
+    auto & mypc = warpx.GetPartContainer();
+
+    /// get number of species (int)
+    auto species_number = mypc.nSpecies();
+
+    /// get species names (std::vector<std::string>)
+    auto species_names = mypc.GetSpeciesNames();
+
+    // write header row
+    ofs << "#";
+    ofs << "step";
+    ofs << m_sep;
+    ofs << "time(s)";
+    ofs << m_sep;
+    ofs << "total(J)";
+    for (int i = 0; i < species_number; ++i)
+    {
+        ofs << m_sep;
+        ofs << species_names[i]+"(J)";
+    }
+    ofs << std::endl;
+}
+///< end constructor
+
+/// destructor
+ParticleMeanEnergy::~ParticleMeanEnergy ()
+{}
+///< end destructor
+
+/// function that computes kinetic energy
+void ParticleMeanEnergy::ComputeDiags (int step)
+{
+
+    /// Judge if the diags should be done
+    if ( (step+1) % m_freq != 0 ) { return; }
```

Comment:
If the user gives `my_diag. frequency = 10` will the diags be written for iterations 9, 19 etc.??

Reply:
If user gives 10, it will be 10, 20, 30, ...
I will also add `step = 0` output.